### PR TITLE
Stock sync

### PR DESCRIPTION
**Pull request by DanielePalombo**
When a shipwire's webhook calls 'stock_controller/create' the action
updates `count_on_hand` in `stock_item` and adds a `stock_movement` to
keep track movements.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,3 +1,9 @@
 Spree::Core::Engine.routes.draw do
   # Add your extension routes here
+  namespace :shipwire_webhooks do
+    resources :stock, only: :create
+
+    # match all HEAD from shipwire for webhook subscription
+    match '*path' => '/spree/shipwire_webhook#subscribe', :via => :head
```
Comment (kennyadsl) [inline config/routes.rb]: `via: :head`

File: app/controllers/spree/shipwire_webhook_controller.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
+class Spree::ShipwireWebhookController < ActionController::Base
+  respond_to :json
+
+  before_action :validate_key, except: :subscribe
```
Comment (kennyadsl) [inline app/controllers/spree/shipwire_webhook_controller.rb]: this is done except `subscribe` but  `subscribe` seems to be the only action here. Am I missing something?

File: lib/solidus_shipwire/engine.rb [hunk 3]
```diff
@@ -10,6 +11,14 @@ class Engine < Rails::Engine
       g.test_framework :rspec
     end
 
+    initializer 'solidus_shipwire.environment', before: 'spree.environment' do
+      Spree::SolidusShipwireConfig = Spree::SolidusShipwireConfiguration.new
```
Comment (kennyadsl) [inline lib/solidus_shipwire/engine.rb]: why not use `Spree::ShipwireConfig`? If one day`Spree` namespace will be converted into `Solidus` we'll have `Solidus::SolidusShipwireConfig`

File: lib/spree/solidus_shipwire_configuration.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  class SolidusShipwireConfiguration < Preferences::Configuration
+    preference :username,     :string,  default: nil
+    preference :password,     :string,  default: nil
+    preference :open_timeout, :integer, default: 2
+    preference :timeout,      :integer, default: 5
+    preference :endpoint,     :string,  default: 'http://api.shipwire.com'
+    preference :logger,       :string,  default: false
+    preference :secret,       :string,  default: nil
+
+    def set_shipwire
```
Comment (kennyadsl) [inline lib/spree/solidus_shipwire_configuration.rb]: maybe `setup_shipwire` sounds better.

File: spec/controllers/shipwire_webhook_spec.rb [hunk 5]
```diff
@@ -0,0 +1,33 @@
+RSpec.describe Spree::ShipwireWebhookController, type: :controller do
+  controller Spree::ShipwireWebhookController do
+    def create
```
Comment (kennyadsl) [inline spec/controllers/shipwire_webhook_spec.rb:3]: should this be `subscribe`?

File: app/controllers/spree/shipwire_webhook_controller.rb (hunk 2 above)
Comment (DanielePalombo) [reply app/controllers/spree/shipwire_webhook_controller.rb]: This controller can be extended by others controller. All actions need to be validated before execute

Comment (kennyadsl) [reply app/controllers/spree/shipwire_webhook_controller.rb]: shouldn't be the others controllers to take care of not running that callback?

Comment (mtylty) [reply app/controllers/spree/shipwire_webhook_controller.rb]: I think this looks good. It's a basic check it shouldn't hurt other code and it's covered with specs.

File: spec/controllers/shipwire_webhook_spec.rb (hunk 5 above)
Comment (mtylty) [reply spec/controllers/shipwire_webhook_spec.rb:3]: I think the point here is testing exactly an action not called `subscribe`.

Anyway I think `subscribe` specs are missing?

Comment (DanielePalombo) [reply spec/controllers/shipwire_webhook_spec.rb:3]: added in https://github.com/nebulab/solidus_shipwire/blob/stock-sync/spec/requests/spree/shipwire_webhook_spec.rb#L1

**Comment by DanielePalombo** [comment]
Can I merge this PR?

**Pull request merged**